fetch_napari_data.py

```diff
@@ -292,68 +263,125 @@ def build_plugins_dataframe() -> pd.DataFrame:
     """
-    summary_df = get_plugin_summary(API_SUMMARY_URL)
-    conda_name_map = fetch(API_CONDA_MAP_URL)
+    extended_summary = api_client.fetch_summary()
+
+    with ThreadPoolExecutor(max_workers=MAX_WORKERS) as executor:
+        all_plugin_data = executor.map(
+            get_plugin_page_data_from_api,
+            extended_summary,
+        )
+
+    def key(p: PluginPageData) -> str:
+        return p.modified_at or ""
+
+    return sorted(
+        all_plugin_data,
+        key=key,
+        reverse=True,
+    )
+
+
+def get_plugin_page_data_from_api(plugin_summary_data):
+    plugin_normalized_name = plugin_summary_data.get("normalized_name")
+    manifest_info = api_client.fetch_manifest(plugin_normalized_name)
 
-    all_plugin_data = []
+    # some PyPI info is not included in the manifest
+    pypi_info = api_client.fetch_pypi_info(plugin_normalized_name)
+    # this is much of the same as what's in the pypi_info
+    package_metadata = manifest_info.get("package_metadata", {})
 
-    def process_plugin(plugin):
-        plugin_data = plugin.copy()
-        plugin_normalized_name = plugin.get("normalized_name")
+    # conda_info is not currently used
+
+    name = manifest_info.get("name", plugin_summary_data.get("name", ""))
+    display_name = manifest_info.get(
+        "display_name", plugin_summary_data.get("display_name", "")
+    )
+    # releases are sorted in descending order
+    plugin_first_release = plugin_summary_data["pypi_versions"][-1]
+    plugin_latest_release = plugin_summary_data["pypi_versions"][0]
+    initial_release_date = get_version_release_date(pypi_info, plugin_first_release)
+    last_updated_date = get_version_release_date(pypi_info, plugin_latest_release)
+    authors, emails = get_authors_and_emails(package_metadata)
+    package_license = get_license(package_metadata)
+    home_pypi = pypi_info.get(
+        "home_page",
+        f"https://pypi.org/project/{plugin_normalized_name}/",
+    )
+    home_github, home_other = get_project_home_url(plugin_summary_data)
+
+    contributions = manifest_info.get("contributions", {})
+    reader_patterns = list(
+        set(
+            ext
+            for reader in contributions["readers"] or []
+            for ext in reader["filename_patterns"]
+        )
+    )
 
-        # Fetch and flatten Conda info and Manifest
-        conda_info = (
-            fetch_conda(plugin_normalized_name)
-            if conda_name_map[plugin_normalized_name] is not None
-            else {}
+    writer_extensions = list(
+        set(
+            ext
+            for writer in contributions["writers"] or []
+            for ext in writer["filename_extensions"]
         )
-        manifest_info = fetch_manifest(plugin_normalized_name)
-        # need pypi info to get the initial and latest release date
-        pypi_info = fetch(urljoin(API_PYPI_BASE_URL, plugin_normalized_name))
-
-        expand_proj_url(plugin_data)
-
-        if conda_info:
```

Comment:
I know we're not using conda info at the moment, so I'm ok to remove it here, but I think it would be nice to include it at some stage. Opened an issue to track